### ISSUE TITLE
Exclude child elements in query results

### ISSUE THREAD

**Issue by uzaysan** [parse-community/parse-server#7030]
**Is your feature request related to a problem? Please describe.**
In parse server we can put parse objects to another parse object. And we can get them with include method.
We also select specific fields from child objects with select method.
But we cannot exclude fields from child object. Lets assume we have Book object. And author field is a pointer to a user. If I want to get author alongside with book object, user object is fully fetched. But If I want to exclude some sensitive fields such as email field, I can't do that. Exclude method only works for Book object. But User objects is ignored.

**Describe the solution you'd like**
We should be able to exclude unwanted fields from child objects. Here is an example code:

```
var query = New Parse.Query("Book");
query.include("author");
query.exclude("someBookField","author.email");
...
const results = await.query.find();
//There is no email field in results.
```

**Describe alternatives you've considered**
We can convert parse objects to JSON and we can delete the unwanted fields by manually. But this is useless if our main porpouse is save bandwidth costs.

**Comment by mtrezza** [comment]
Thanks for this suggestion; initially [discussed](https://community.parseplatform.org/t/exclude-doesnt-work-for-child-objects/1068/3) in Parse Forum.

Would you want to open a PR and create a failing test, so that we can get started to fix this?
You can take a look at the [contribution guide](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md) for how to set up Parse Server for local debugging.

The basic steps are:
1. Fork Parse Server and clone repository locally.
2. Create new branch.
3. Look at `ParseQuery.spec.js` where there is already a test for `exclude`:
https://github.com/parse-community/parse-server/blob/568c285369717ef5423e63304291016b41daca5d/spec/ParseQuery.spec.js#L4755-L4766
4. Add a test for excluding a child element; using the existing `exclude` test as guidance. If the test fails, you can debug the test and look at where it fails. It could be somewhere in the Parse JS SDK or in Parse Server.
5. Add the fix in Parse Server or the Parse JS SDK to make the test pass.

If you need any guidance please let us know and we'll be glad to support.

**Comment by dplewis** [comment]
@cbaker6 Since you are already looking into exclude can you look into this one?

**Comment by mtrezza** [comment]
As discussed in https://github.com/parse-community/parse-server/issues/7390#issuecomment-845083794, the PR for this issue seems to be mixed into PR https://github.com/parse-community/parse-server/pull/7242.

I am glad to see the issue addressed. Having said that, for future PRs we should try to keep distinct issues and corresponding distinct PRs, unless issues are so intertwined that they cannot be addressed separately. Otherwise PRs blow up in scope and we increase the risk of bugs.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)
